library/core/src/ptr/alignment.rs

```diff
@@ -118,10 +121,39 @@ impl Alignment {
     /// assert_eq!(Alignment::new(1024).unwrap().log2(), 10);
     /// ```
     #[unstable(feature = "ptr_alignment_type", issue = "102070")]
+    #[rustc_const_unstable(feature = "ptr_alignment_type", issue = "102070")]
     #[inline]
-    pub fn log2(self) -> u32 {
+    pub const fn log2(self) -> u32 {
         self.as_nonzero().trailing_zeros()
     }
+
+    /// Returns a bit mask that can be used to match this alignment.
+    ///
+    /// This is equivalent to `!(self.as_usize() - 1)`.
+    ///
+    /// # Examples
+    ///
+    /// ```
+    /// #![feature(ptr_alignment_type)]
+    /// #![feature(ptr_mask)]
+    /// use std::ptr::{Alignment, NonNull};
+    ///
+    /// let one = <NonNull<u8>>::dangling().as_ptr();
+    /// let four = <NonNull<u32>>::dangling().as_ptr();
+    ///
+    /// assert_eq!(four.mask(Alignment::of::<u8>().mask()), four);
+    /// assert_eq!(four.mask(Alignment::of::<u16>().mask()), four);
+    /// assert_eq!(four.mask(Alignment::of::<u32>().mask()), four);
+    /// assert_ne!(four.mask(Alignment::of::<u64>().mask()), four);
+    /// assert_eq!(one.mask(Alignment::of::<u64>().mask()), one);
+    /// ```
+    #[unstable(feature = "ptr_alignment_type", issue = "102070")]
+    #[rustc_const_unstable(feature = "ptr_alignment_type", issue = "102070")]
+    #[inline]
+    pub const fn mask(self) -> usize {
+        // SAFETY: The alignment is always nonzero, and therefore decrementing won't overflow.
+        !(unsafe { self.as_usize().unchecked_sub(1) })
```

Comment:
Sadly, `unchecked_sub(1)` is useless on unsigned types <https://github.com/llvm/llvm-project/issues/53377>

I guess it's fine to leave it, though, since it's clearly correct.  And trying to do it in `isize` would be UB.

Reply:
Oh, that's unfortunate. I even was considering at some point adding a `decrement()` method to `NonZeroU*` that could use `unchecked_sub` since it would never overflow, but I guess that that also wouldn't work.